src/Context.php

```diff
@@ -101,7 +108,7 @@ public function getNavigator(): GraphNavigatorInterface
         return $this->navigator;
     }
 
-    public function getExclusionStrategy(): ?ExclusionStrategyInterface
+    public function getExclusionStrategy(): ?DisjunctExclusionStrategy
```

Comment:
If we change this return type, then "we delegate" the control of what interfaces the "exclusionStrategy" property could contain. 

In this case i extended it with a new interface "ValueExclusionStrategyInterface" that defines a method which accepts a `value` also, beside the `metadata` and `context`